qdrant_client/local/local_collection.py

```diff
@@ -702,20 +772,61 @@ def recommend(
         )
 
         if strategy == types.RecommendStrategy.AVERAGE_VECTOR:
-            query_vector = self._recommend_average(
-                positive_vectors,
-                negative_vectors,
-            )
+            # Validate input
+            if positive_vectors:
+                query_vector = self._recommend_average_dense(
+                    positive_vectors,
+                    negative_vectors,
+                )
+            elif sparse_positive_vectors:
+                query_vector = self._recommend_average_sparse(
+                    sparse_positive_vectors,
+                    sparse_negative_vectors,
+                )
+            else:
+                raise ValueError("No positive examples given")
+
         elif strategy == types.RecommendStrategy.BEST_SCORE:
-            query_vector = RecoQuery(
-                positive=positive_vectors,
-                negative=negative_vectors,
-            )
+            if positive_vectors or negative_vectors:
+                query_vector = RecoQuery(
+                    positive=positive_vectors,
+                    negative=negative_vectors,
+                )
+            else:
+                query_vector = SparseRecoQuery(
+                    positive=sparse_positive_vectors, negative=sparse_negative_vectors
+                )
```

Comment:
~what if there are no `sparse_positive_vectors` nor `sparse_negative_vectors` either?~
Edit: it is already checked in https://github.com/qdrant/qdrant-client/blob/164c8375334510279e38cc1ca4912ac892705ab7/qdrant_client/local/local_collection.py#L676